float64[] goal_vector
